Classes/Controller/RoutingController.php

```diff
@@ -176,10 +176,15 @@ protected function getControllerParameters($subroute, $extensionKey = null)
                 $namespaceParts = explode('.', $controllerParameters['@package']);
                 if (count($namespaceParts) === 2) {
                     $controllerParameters['@vendor'] = $namespaceParts[0];
-                    $controllerParameters['@extension'] = GeneralUtility::underscoredToUpperCamelCase($namespaceParts[1]);
+                    $controllerParameters['@extension'] = $namespaceParts[1];
                 } else {
-                    $controllerParameters['@extension'] = GeneralUtility::underscoredToUpperCamelCase($namespaceParts[0]);
+                    $controllerParameters['@extension'] =$namespaceParts[0];
                 }
+                
+                $controllerParameters['@extension'] = ucfirst($controllerParameters['@extension']) !== $controllerParameters['@extension'] ? 
+                    GeneralUtility::underscoredToUpperCamelCase($namespaceParts[0]) : 
+                    $controllerParameters['@extension'];
+                
```

Comment:
would prefer something like

```
if (ucfirst($controllerParameters['@extension']) !== $controllerParameters['@extension']) {
    $controllerParameters['@extension'] = GeneralUtility...
}
```

since the tested case is more obvious but then I wonder if this could only happen in a context of a non-namespaced extension, and as such it could be possibly be better integrated into the previous test/assignment (second / `else` branch)